tests/run-make/redundant-libs/rmake.rs

```diff
@@ -0,0 +1,29 @@
+// rustc will remove one of the two redundant references to foo below.  Depending
+// on which one gets removed, we'll get a linker error on SOME platforms (like
+// Linux).  On these platforms, when a library is referenced, the linker will
+// only pull in the symbols needed _at that point in time_.  If a later library
+// depends on additional symbols from the library, they will not have been pulled
+// in, and you'll get undefined symbols errors.
+//
+// So in this example, we need to ensure that rustc keeps the _later_ reference
+// to foo, and not the former one.
+
+//@ ignore-cross-compile
+// Reason: the compiled binary is executed
+
+use run_make_support::{
+    build_native_dynamic_lib, build_native_static_lib, cwd, is_msvc, rfs, run, rustc,
+};
+
+fn main() {
+    build_native_dynamic_lib("foo");
```

Comment:
Problem: on windows msvc *specifically*, `build_native_dynamic_lib` only produces `foo.dll`, which is the dynamic library. But for the purposes of linking via `link.exe`, it only accepts the *import library for the dynamic library*, i.e. `foo.dll.lib`.

Trying to produce the import library for the dynamic library is... not quite trivial:  see https://stackoverflow.com/questions/9360280/how-to-make-a-lib-file-when-have-a-dll-file-and-a-header-file.

There are two possible paths here:

1. We could try to use the `cc` crate or otherwise try to make our own import library with `dumpbin`, `lib` and some clever manipulation...
2. Or we could just `ignore-msvc` with reason "msvc linking to dylib requires building an import library for the dylib, left as an exercise for the future reader".

I'm personally inclined to go with option (2), unless you want to tackle this lol.